src/duration.rs

```diff
@@ -443,14 +443,42 @@ pub struct Generic<T> {
     scaling_factor: Fraction,
 }
 
+impl<T: TimeInt> Generic<T> {
+    /// Try to create a new, equivalent `Generic` with the given _scaling factor_
```

Comment:
Just to help make what is going on here a little clearer:
```suggestion
    /// Try to create a new, equivalent `Generic` with the given _scaling factor_.
    ///
    /// To minimise errors due to truncation rounding, the new integer component is
    /// calculated as `self.integer * (self.scaling_factor / scaling_factor)`.
```

Reply:
I have stopped contributing here and created [`fugit`](https://crates.io/crates/fugit) instead.